mvcc/kvstore.go

```diff
@@ -372,16 +369,12 @@ func (s *store) restore() error {
 	if len(finishedCompactBytes) != 0 {
 		s.compactMainRev = bytesToRev(finishedCompactBytes[0]).main
 
-		if s.lg != nil {
-			s.lg.Info(
-				"restored last compact revision",
-				zap.String("meta-bucket-name", string(metaBucketName)),
-				zap.String("meta-bucket-name-key", string(finishedCompactKeyName)),
-				zap.Int64("restored-compact-revision", s.compactMainRev),
-			)
-		} else {
-			plog.Printf("restore compact to %d", s.compactMainRev)
-		}
+		s.lg.Info(
```

Comment:
@jingyih - actually, one comment - don't we need to set the log for the store`s.lg = zap.NewNop()` ?

Reply:
This is a good question. In etcd we use the store created by `newWatchableStore()`. I made sure the logger is not nil in that function, therefore s.lg will not be nil here. But I should apply your suggestion for extra safety.

Reply:
Fixed here: https://github.com/etcd-io/etcd/blob/6b389bf23ce298f6d33bb9a2b85e3873609dc440/mvcc/kvstore.go#L116

Reply:
Awesome! Thanks @jingyih